compendium/ServiceComponent/include/cppmicroservices/servicecomponent/detail/ComponentInstanceImpl.hpp

```diff
@@ -448,7 +448,10 @@ namespace cppmicroservices
                                       "and/or the service implementation does not implement all of the "
                                       "service interface's methods. A default constructor or a constructor "
                                       "with an AnyMap input parameter for the configuration properties is "
-                                      "required when inject-references is false. ");
+                                      "required when inject-references is false. This error can also be "
+                                      "triggered when inheriting from "
+                                      "cppmicroservices::service::component::usrDefinedMethodAssertion "
+                                      "without properly defining the Modified, Activate, and Deactivate methods");
```

Comment:
I would personally work on this error message; I think having very clear errors is important to usability.

The first part of this message is written in terms of a solution ("a default constructor with ... is required"), but the new part is written in terms of the problem, and is self-referential ("This error can also be triggered when"). Changing perspective partway through the message is a bit confusing. The phrasing also suggests that the second part describes a cause that has not been mentioned before, even though the first sentence, as I understand it, describes the general case for this situation. I had to read it a couple times to make this connection.

(As someone who has not encountered the conditions that would cause this error) I also wonder if there is something special about usrDefinedMethodAssertion, or if this could be generalized to remind the user that they may need to implement additional methods if they are inheriting from a class with virtual members.

Reply:
I agree. I updated this